### DISCUSSION TITLE
How to show axis ticks when plotting graphs?

### DISCUSSION THREAD

**Discussion by jaredstrader** [alandefreitas/matplotplusplus#128] [Q&A] [answered]
I am trying to plot a graph where each vertex is associated with a specific coordinate. From my understanding, the ways to do this are 1) make a scatter plot then draw lines between the coordinates or 2) plot as a graph by setting the x_data and y_data of the graph object. I opted for the second approach.

Here is a minimum working example:

```
#include <matplot/matplot.h>

int main( int argc, char** argv )
{
  std::vector<std::pair<size_t,size_t>> edges = {{0,1},{0,2},{0,3},{1,2},{1,3},{2,3}};
  std::vector<double> x_data = { 0.2, -0.1, 1.1, 1.3};
  std::vector<double> y_data = {-0.1,  0.8, 0.3, 1.2};

  auto g = matplot::graph(edges);
  g->show_labels(false);
  g->x_data(x_data);
  g->y_data(y_data);

  matplot::axis(matplot::square);
  matplot::xlabel("X");
  matplot::ylabel("Y");
  matplot::show();

  return 0;
}
```

However, the axes ticks do not appear in the plot, so my question is how to show the axis ticks? I tried searching the examples as well as searching the code, but I cannot find how to enable the axis ticks.

**Comment by alandefreitas** [comment, marked as answer]
As a [graph drawing](https://en.wikipedia.org/wiki/Graph_drawing) is not the same as the graph itself, a network plot only keeps a collection of vertex pairs that get ultimately represented as scatter plots and lines with a reasonable [specific layout](https://github.com/alandefreitas/matplotplusplus/blob/e3e48e25505fd5b6cff4d53b0c33c29dcaf9f88e/source/matplot/axes_objects/network.h#L21). The specific vertex positions stored in the network object only as pre-processed data to make things faster, but they are not part of the interface, and they can change at any time.

Each plot type adjusts the axes to match what's expected for that plot type unless there are more plots in the same axes, in which case some heuristics decide what's best for that plot. In the case of graphs, the ticks are usually hidden because that would confuse the graph concept with the graph drawing, whose absolute positions not meaningful unless we're discussing the graph drawing. 

- If you want each vertex associated with a fixed specific coordinate, then you need is a scatter plot and some lines. In that case, you are plotting something that's also a network/graph, but not only a network/graph.
- If you want a new layout algorithm, then you need to implement this algorithm to extend the ones in `network.h` or simply create scatter plots and lines directly. 
- If you only want the axes ticks to appear in the plot, regardless of where the vertices are positioned, then you can get a handle to the current axes with `matplot::gca()` and turn the [ticks back on](https://github.com/alandefreitas/matplotplusplus/blob/e3e48e25505fd5b6cff4d53b0c33c29dcaf9f88e/source/matplot/core/axes_type.h#L94).

**Reply by jaredstrader** [reply]
Thanks for the detailed explanation! I will use a scatter plot and lines for now; however, I may look into `network.h` later.
